test/runtests.jl

```diff
@@ -19,13 +19,80 @@ abstract type Iterable end
     Base.IteratorEltype(::Type{Iterable})::Union{Base.HasEltype, Base.EltypeUnknown}
 
     if Base.IteratorEltype(Iterable) == Base.HasEltype()
-        eltype(::Iterable)
+        eltype(::Iterable) || eltype(::Type{Iterable})
     end
 
 end
 
-@test Interfaces.implements(Int, Iterable; mods=[Base])
+@test Interfaces.implements(Int, Iterable, [Base])
+
+abstract type Indexable end
+
+@interface Indexable begin
+    getindex(::Indexable, i)
+    setindex!(::Indexable, v, i)
+    firstindex(::Indexable)
+    lastindex(::Indexable)
+end
+
+@test Interfaces.implements(Array, Indexable, [Base])
+
+@interface AbstractArray begin
+
+    <: AbstractArray
```

Comment:
i think this bit of syntax just looks wrong... i think we might be able to come up with something better 🤔